pkg/extensions/sync/sync_internal_test.go

```diff
@@ -168,6 +168,37 @@ func TestSyncInternal(t *testing.T) {
 		So(err, ShouldNotBeNil)
 	})
 
+	Convey("Test filterRepos()", t, func() {
+		repos := []string{"repo", "repo1", "repo2", "repo/repo2", "repo/repo2/repo3/repo4"}
+		contents := []Content{
+			{
+				Prefix: "repo*",
+			},
+			{
+				Prefix: "/repo/**",
+			},
+			{
+				Prefix: "repo",
+			},
+		}
+		filteredRepos, err := filterRepos(repos, contents)
+		So(err, ShouldBeNil)
+		So(filteredRepos[0], ShouldResemble, []string{"repo", "repo1", "repo2"})
+		So(filteredRepos[1], ShouldResemble, []string{"repo/repo2", "repo/repo2/repo3/repo4"})
+
+		// last prefix shouldn't match, because it was already matched on the first prefix (repo*)
+		So(len(filteredRepos), ShouldEqual, 2)
```

Comment:
Maybe it makes sense to reorder prefixes and add new tests.
If I get it right, if we interchange lines 175 with 181 , than the length(filteredRepos) should be 3, right?

Reply:
Yes, that's right.